blazar/manager/enforcement.py

```diff
@@ -195,28 +210,23 @@ def check_lease_duration(self, lease_values, lease=None):
                     lease_duration = end_date - now
 
         lease_duration = lease_duration.total_seconds()
-
-        lease_exception = self.get_lease_exception(user_name)
-        # A one-time lease exception can be set for the user
-        if lease_exception is not None:
-            lease_exception = int(lease_exception)
-            if lease_duration > lease_exception:
-                raise common_ex.NotAuthorized(
-                    'Requested lease to last %d seconds, which is longer than '
-                    'maximum allowed of %d seconds for user %s' %
-                    (lease_exception, user_name))
-        elif project_enforcement_id in self.project_max_lease_durations:
-            project_max_lease_duration = self.project_max_lease_durations.get(
-                project_enforcement_id)
-            if project_max_lease_duration != -1:
-                if (lease_duration) > project_max_lease_duration:
+        lease_violates_max_duration = (
+            CONF.enforcement.default_max_lease_duration != -1 and
+            lease_duration > CONF.enforcement.default_max_lease_duration)
+
+        if (lease_violates_max_duration and
```

Comment:
I think some early returns can help clean up this function further.

```python

if not lease_violates_max_duration:
    return True

if self.is_exempt_project(lease_values['project_id']):
    return True

user_name = self._get_user_name(lease_values['user_id'])
lease_exception = self.get_lease_exception(user_name)

if not lease_exception:
    raise common_ex.NotAuthorized('Lease is longer than...')

if lease_duration > int(lease_exception):
    raise common_ex.NotAuthorized('Requested lease to last...')

return True
```

This makes it have much less nesting, and maybe is easier to read? I find early returns a very nice tool for long term code quality, but there is usually balance needed. This chain of checks seems like a nice opportunity though.

Reply:
Yea, this looks cleaner. Updated.